src/gossip_agent/waku_handling.rs

```diff
@@ -357,27 +379,53 @@ pub async fn check_message_validity<
 }
 
 /// Check for peer connectivity, try to reconnect if there are disconnected peers
-pub fn network_check(node_handle: &WakuNodeHandle<Running>) -> Result<(), Box<dyn Error>> {
+pub fn network_check(node_handle: &WakuNodeHandle<Running>) -> Result<(), anyhow::Error> {
     let binding = node_handle
         .peer_id()
         .expect("Failed to get local node's peer id");
     let local_id = binding.as_str();
 
-    node_handle
-        .peers()?
-        .iter()
-        // filter for nodes that are not self and disconnected
-        .filter(|&peer| (peer.peer_id().as_str() != local_id) & (!peer.connected()))
-        .map(|peer: &WakuPeerData| {
-            debug!("Disconnected peer data: {:#?}", &peer);
-            node_handle.connect_peer_with_id(peer.peer_id().to_string(), None)
-        })
-        .for_each(|res| {
-            if let Err(x) = res {
-                warn!("Could not connect to peer: {}", x)
-            }
-        });
-    Ok(())
+    let _peers = node_handle.peers();
+
+    match node_handle.peers() {
+        Ok(peers) => {
+            peers
+                .iter()
+                // filter for nodes that are not self and disconnected
+                .filter(|&peer| (peer.peer_id().as_str() != local_id) & (!peer.connected()))
+                .map(|peer: &WakuPeerData| {
+                    debug!("Disconnected peer data: {:#?}", &peer);
+                    node_handle.connect_peer_with_id(peer.peer_id().to_string(), None)
+                })
+                .for_each(|res| {
+                    if let Err(x) = res {
+                        warn!("Could not connect to peer: {}", x)
+                    }
+                });
+            Ok(())
+        }
+        Err(e) => Err(anyhow!(e)),
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum FilterPeerSubscriptionsError {
```

Comment:
These look great, I wonder if they can be combined in to a bigger Error type like `WakuHandlingError`?